server/runtime.go

```diff
@@ -37,16 +40,111 @@ func (s *Server) Version(ctx context.Context, req *pb.VersionRequest) (*pb.Versi
 func (s *Server) CreatePodSandbox(ctx context.Context, req *pb.CreatePodSandboxRequest) (*pb.CreatePodSandboxResponse, error) {
 	var err error
 
-	// TODO: Parametrize as a global argument to ocid
-	ocidSandboxDir := "/var/lib/ocid/sandbox"
-	podSandboxDir := filepath.Join(ocidSandboxDir, req.GetConfig().GetName())
+	// process req.Name
+	name := req.GetConfig().GetName()
+	if name == "" {
+		d, err := ioutil.TempDir(s.sandboxDir, "")
+		if err != nil {
+			return nil, err
+		}
+		name = filepath.Base(d)
+	}
+	podSandboxDir := filepath.Join(s.sandboxDir, name)
+	if err := os.MkdirAll(podSandboxDir, 0755); err != nil {
+		return nil, err
+	}
 
+	// creates a spec Generator with the default spec.
 	g := generate.New()
 
-	// TODO: Customize the config per the settings in the req
+	// process req.Hostname
+	hostname := req.GetConfig().GetHostname()
+	if hostname != "" {
+		g.SetHostname(hostname)
+	}
+
+	// process req.LogDirectory
+	logDir := req.GetConfig().GetLogDirectory()
+	if logDir == "" {
+		logDir = "/var/log/ocid/pods"
+	}
+	fmt.Println(logDir)
+
+	// TODO(hmeng): how to utilize this info?
+	dnsOpts := req.GetConfig().GetDnsOptions()
+	fmt.Println(dnsOpts)
+
+	// TODO(hmeng): how to utilize this info?
+	portMappings := req.GetConfig().GetPortMappings()
+	fmt.Println(portMappings)
+
+	// TODO(hmeng): the unit of cpu here is core. How to map it into specs.Spec.Linux.Resouces.CPU?
+	// TODO(hmeng): ocitools/generate currently does not support setting specs.Spec.Linux.Resouces.CPU?
+	cpu := req.GetConfig().GetResources().GetCpu()
+	if cpu != nil {
+		limits := cpu.GetLimits()
+		requests := cpu.GetRequests()
+		fmt.Println(limits)
+		fmt.Println(requests)
+	}
+
+	// TODO(hmeng): ocitools/generate currently does not support setting specs.Spec.Linux.Resouces.Memory?
+	// Should I add functions into ocitools/generate?
+	memory := req.GetConfig().GetResources().GetMemory()
+	if memory != nil {
+		// limits sets specs.Spec.Linux.Resouces.Memory.Limit
+		limits := memory.GetLimits()
+		// requests sets specs.Spec.Linux.Resouces.Memory.Reservation
+		requests := memory.GetRequests()
+		fmt.Println(limits)
+		fmt.Println(requests)
+	}
+
+	// TODO(hmeng): Should Server maintain these labels?
```

Comment:
Okay. Then I will add a new field into `Server`, `podLabels`:
`podLabels map[string]map[string]string`, the key will be pod id, the value will be the labels.